pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go

```diff
@@ -446,8 +446,14 @@ func TestWithBinding(t *testing.T) {
 				scLister,
 			}
 			gotStatus := p.Filter(context.Background(), nil, test.Pod, node)
-			if !reflect.DeepEqual(gotStatus, test.wantStatus) {
-				t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)
+			if test.wantStatus.Code() == framework.Error {
+				if diff := cmp.Diff(test.wantStatus.Reasons(), gotStatus.Reasons()); diff != "" {
```

Comment:
Is it worth to duplicate lots of logic of `volumezone.Filter()` in a mock plugin to remove such workaround? WDYT @alculquicondor